### ISSUE TITLE
Limit number of caches for "show on map"

### ISSUE THREAD

**Issue by Bananeweizen** [cgeo/cgeo#7521]
##### Detailed steps causing the problem:
- Have a stored list of caches containing > 1000 caches (e.g. all caches of a small country)
- Use "Show on map" toolbar button.

##### Request
The map is unusable, and the user has to zoom to actually make use of this. We discussed quite a while ago to limit the zoom factor for this action to a hard number, but didn't come to a conclusion. By now I want to suggest to limit the number of shown caches instead: Calculate the 50 caches closest to current user position, and open the map with a rectangle containing those 50 caches. In practice this will lead to something between 50 and 200 caches to be shown, since the map implementation has to use an integer zoom factor that definitely contains the rectangle, and in most cases this shown area will contain _more_ than the wanted rectangle.
I do _not_ propose to limit what is shown on the map as such (e.g. if you move the map around, you will still see the other 1000+ caches, just not initially).

Basically I suggest that we do in code what the user would normally do manually with the current implementation: Zoom in until the icons no longer "overflow" the complete map.

**Comment by Lineflyer** [comment]
We do now also have the compact icons, which makes the map quite readable again.

**Comment by moving-bits** [comment]
I do like the proposal by @Bananeweizen - it's a situation I also encounter frequently, and I also zoom in until I get a reasonable amount of caches show.

> We do now also have the compact icons, which makes the map quite readable again.
> So I would rather think about a better mechanism to (auto?) enable the compact icons depending on cache density.

I'm unsure about this one. I fear that it may be confusing for our user (at least for a while, where compact icons is new), and also it does not really solve the problem described in this issue, it just sets the limit higher. - OTOH this is the way gc.com handles its map, so gc.com users are sort of accustomed to it.

A different approach could be to distinguish between the two icon modes and set the limit accordingly: If in normal icons mode, set the zoom to get something between 50 and 200 caches visible in the viewport, and if compact icons mode is active, zoom to maybe twice or three times this amount (values chosen randomly).

**Comment by Lineflyer** [comment]
I should have read the issue from @Bananeweizen more carefully. I thought the other caches would be not shown...but in fact we would only  limit the initial zoom. So forget my proposal of the compact icons for this case...its another separate topic then.

But still: Using 50 and using those next to user position is probably not matching the expectation of all users/usecases. So I am still undecided about a good implementation.

**Comment by Bananeweizen** [comment]
As long as your position _is_ inside the bounding box of all caches of that list, what else would you expect? If the user position is not inside the bounding box, we can of course remain with the old behavior of showing all caches (and do not need to select an artificial "border region" of 50 caches which are most close to the user).
